std/typecons.d

```diff
@@ -149,25 +157,37 @@ public:
     ~this()
     {
         debug(Unique) writeln("Unique destructor of ", (_p is null)? null: _p);
-        if (_p !is null) delete _p;
-        _p = null;
+        release();
     }
-    /** Returns whether the resource exists. */
-    @property bool isEmpty() const
-    {
-        return _p is null;
-    }
-    /** Transfer ownership to a $(D Unique) rvalue. Nullifies the current contents. */
-    Unique release()
+
+    /** Frees the underlying $(D RefT) and nulls it */
+    void release()
     {
         debug(Unique) writeln("Release");
-        auto u = Unique(_p);
-        assert(_p is null);
-        debug(Unique) writeln("return from Release");
-        return u;
+        if (_p !is null)
+        {
+            import core.memory : GC;
+
+            destroy(_p);
+            GC.free(cast(void*)_p);
```

Comment:
Only the compiler can get the context pointer, and it will only do so when you initialize a nested struct (the normal way, not by emplacing).
```d
unique(T, Args...)(Args args)
{
   auto tmp = T(args); // will init context pointer
   moveEmplace(storage, tmp);
}
```